fetch_news.py

```diff
@@ -0,0 +1,360 @@
+                force=force,
+                entities=entities,
+                stemmed_entities=stemmed_entities,
+                stemmed_harm_keywords=stemmed_harm_keywords,
+            ):
+                last_hit = now
+
+def get_entities(mongo_client = None, connection_string = None):
+    if not mongo_client:
+        mongo_client = get_mongo_client(connection_string)
+    if not mongo_client:
+        raise Exception("No connection string provided")
+    incidents_collection = mongo_client['aiidprod'].incidents
+
+    entity_fields = [
+        'Alleged deployer of AI system',
+        'Alleged developer of AI system',
+        'Alleged harmed or nearly harmed parties',
+    ]
+
+    entities = set()
+
+    projection = {}
+    for field in entity_fields:
+        projection[field] = True
+
+    for incident in incidents_collection.find({}, projection):
+        for field in entity_fields:
+            for entity in incident[field]:
+                if not entity in ignored_entities:
+                    entities.add(entity)
+
+    return entities 
+
+def get_mongo_client(connection_string = None):
+    if not connection_string:
+        connection_string = environ.get('MONGODB_CONNECTION_STRING')
+    if connection_string:
+        return MongoClient(connection_string)
+ 
+def get_mean_embedding(mongo_client = None, connection_string = None, classification = None):
+    if not mongo_client:
+        mongo_client = get_mongo_client(connection_string)
+    if not mongo_client:
+        raise Exception("No connection string provided")
+    
+    query = { 'embedding': { '$exists': True } }
+
+    if classification:
+        classifications_collection = mongo_client['aiidprod'].classifications
+        classifications_query = {   
+          'namespace': classification[0], 
+          'attributes': { 
+              '$elemMatch': { 
+                  'short_name': classification[1], 
+                  'value_json': {'$regex': classification[2]}
+              }
+          }
+        }
+        incident_ids = [
+            incident['incident_id'] for incident in classifications_collection.find(
+                classifications_query,
+                {'incident_id': True}
+            )
+        ]
+        query['incident_id'] = {'$in': incident_ids}
+
+    incidents_collection = mongo_client['aiidprod'].incidents
+    m = np.array([0] * 768)
+    for offset, incident in enumerate(incidents_collection.find(
+        query, 
+        { 'embedding': { 'vector': True } }
+    )):
+        i = offset + 1
+
+        # As the number of vectors contributing to the mean increases,
+        # the contribution of each one decreases:
+        #
+        # i=1 → m = (0/1)m + (1/1)v
+        # i=2 → m = (1/2)m + (1/2)v
+        # i=3 → m = (1/3)m + (2/3)v
+        # i=4 → m = (3/4)m + (1/4)v
+        # ...
+        v = np.array(incident['embedding']['vector'])
+        m = ( v * 1 / i) + (m * (i - 1) / i)
+
+    return m
+
+
+def process_url(
```

Comment:
In the future, adopt a docstring convention. Example: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html